spec/models/upload_spec.rb

```diff
@@ -78,6 +78,17 @@
     expect(upload.thumbnail_width).to eq(nil)
   end
 
+  it 'returns error when image resolution is to big' do
+    begin
+      SiteSetting.max_image_megapixels = 10
+      upload = UploadCreator.new(huge_image, "image.png").create_for(user_id)
+      expect(upload.id).to be_nil
```

Comment:
```suggestion
      expect(upload.persisted?).to eq(true)
```

Might be clearer this way.